manifests/apache.pp

```diff
@@ -100,6 +103,12 @@
       deploy        => false,
       password_file => $ca_key_password_file,
       build_dir     => $certs::ssl_build_dir,
+    } ->
+    file { "${apache_cert_path}.crt":
+      ensure => file,
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0440',
```

Comment:
What is this for? To satisfy the `require    => File["${apache_cert_path}.crt"],` below? I don't think this will correctly notify when the cert changed.

Reply:
Correct. This was put here to provide something to normalize the difference between when certificates are generated and when they are provided by the user. I felt like the deploy step needed to have some way to ensure that the thing actually existed. 

I could try and enhance https://github.com/theforeman/puppet-certs/blob/master/lib/puppet/type/cert.rb to auto create a file resource, I was trying to avoid too much investment in this provider given it's complexity. My goal being to reduce down the simplicity of how all the cert stuff works step by step.

Reply:
can't we do smth like

```puppet
if $server_cert {
$require_cert = File["${apache_cert_path}.crt"]
} else {
$require_cert = Cert[$apache_cert_name]
}
…
if $deploy {
    certs::keypair { $apache_cert_name:
      require => $require_cert
   }
}
```

Reply:
Good suggestion - added